app/views/static/deploy-provider-details-existing-vms.html.haml

```diff
@@ -0,0 +1,137 @@
+        %button.btn.btn-default.primary-action{"ng-click" => "removeRoles()",
+        "ng-disabled" => "disableMasterNodeActions", :title => "Remove all roles from the selected items", :type => "button"}
+          = _("Remove Roles")
+    .table-responsive
+      %table#DataTables_Table_0.datatable.table.table-striped.table-bordered.dataTable.no-footer{"aria-describedby" => "DataTables_Table_0_info", :role => "grid"}
+        %thead
+          %tr{:role => "row"}
+            %th.checkbox-col{:colspan => "1", :rowspan => "1", :tabindex => "0"}
+              %input{"ng-checked" => "nodeData.allFilterdNodesSelected",
+              "ng-click" => "toggleAllFilteredNodesSelected()",
+              "ng-disabled" => "!nodeData.filteredNodes || nodeData.filteredNodes.length < 1", :type => "checkbox"}/
+            %th{:colspan => "1", :rowspan => "1", :tabindex => "0"}
+              = _("VM/Instance Name")
+            %th{:colspan => "1", :rowspan => "1", :tabindex => "0"}
+              = _("Roles")
+            %th{:colspan => "1", :rowspan => "1", :tabindex => "0"}
+              = _("# CPUS")
+            %th{:colspan => "1", :rowspan => "1", :tabindex => "0"}
+              = _("Memory")
+            %th{:colspan => "1", :rowspan => "1", :tabindex => "0"}
+              = _("Disk Size")
+        %tbody
+          %tr{"ng-class" => "{odd: ($index / 2 == 0), even: ($index / 2 != 0)}", "ng-repeat" => "node in nodeData.filteredNodes track by $index"}
+            %td.checkbox-col
+              %input{"ng-checked" => "node.selected", "ng-click" => "toggleNodeSelected(node)", :type => "checkbox"}/
+            %td {{node.name || node.vmName}}
+            %td
+              .role-column
+                %span.label.label-primary{"ng-if" => "node.master"}
+                  Master
+                  %a
+                    %span.fa.fa-close{"aria-hidden" => "true", "ng-click" => "removeMaster(node)"}
+                %span.label.label-primary{"ng-if" => "node.node"}
+                  Node
+                  %a
+                    %span.fa.fa-close{"aria-hidden" => "true", "ng-click" => "removeNode(node)"}
+                %span.label.label-primary{"ng-if" => "node.storage"}
+                  Storage
+                  %a
+                    %span.fa.fa-close{"aria-hidden" => "true", "ng-click" => "removeStorage(node)"}
+                %span.label.label-primary{"ng-if" => "node.loadBalancer"}
+                  Load Balancer
+                  %a
+                    %span.fa.fa-close{"aria-hidden" => "true", "ng-click" => "removeLoadBalancer(node)"}
+                %span.label.label-primary{"ng-if" => "node.dns"}
+                  DNS
+                  %a
+                    %span.fa.fa-close{"aria-hidden" => "true", "ng-click" => "removeDns(node)"}
+                %span.label.label-primary{"ng-if" => "node.etcd"}
+                  Etcd
```

Comment:
> i18n here and above

@martinpovolny my initial thought was not to use i18n on the roles (Master, Node, Etcd, Load Balancer, Storage, Infra, DNS) since they are terms.
I've seen both you and @himdel commenting on it so I'll add that.